segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/WriteOperation.java

```diff
@@ -126,6 +130,8 @@
                                 return getLastChunk(txn)
                                         .thenComposeAsync(v ->
                                                         writeData(txn)
+                                                                // HACK: delay a while before commit the metadata
+                                                                .thenCompose(vvv -> createDelayFuture(hackDelayMillis))
```

Comment:
```.thenComposeAsync(..., , chunkedSegmentStorage.getExecutor())```
We don't want any storage code executed on ForkJoin thread pool. (dependent actions run on same threadpool that completed the future, so by mistake whole LTS might end up running on ForkJoin bringing down the house)

Reply:
See my previous comment.